bech32-th/test/Codec/Binary/Bech32/THSpec.hs

```diff
@@ -38,9 +39,7 @@ spec =
 
             describe "Parsing invalid human-readable prefixes should fail." $
                 forM_ invalidHumanReadableParts $ \(hrp, expectedError) ->
-                    it (show hrp) $
-                        mkHumanReadablePartExp hrp
-                            `shouldThrow` (== expectedError)
+                    it (show hrp) $ (show <$!!> mkHumanReadablePartExp hrp) `shouldThrow` (== expectedError)
```

Comment:
Could you please force the value in `mkHumanReadablePartExp` and possibly - for clarity - use an auxilliary function? e.g.
```haskell
fullyEvaluateExp :: IO Exp -> IO Exp
```

Reply:
I can't write that function easily because there is no `NFData` instance for `Exp`.

If you don't mine me introducing orphaned instances, I can give this a go.

Reply:
There is no problem just writing it in terms of `show`, but I was suggesting to put this into a separate function.

Reply:
Introduced new `forceViaShowM` function.